File: owner/backend/models/owner.go
package models

import "time"

// Pet представляет питомца владельца (полная структура из PetID)
type Pet struct {
	ID                 int       `json:"id"`
	OwnerID            int       `json:"owner_id"`
	CuratorID          *int      `json:"curator_id,omitempty"`
	OrganizationID     *int      `json:"organization_id,omitempty"`
	Name               string    `json:"name"`
	Species            string    `json:"species"`
	Breed              *string   `json:"breed,omitempty"`
	Age                int       `json:"age"`
	Sex                string    `json:"sex"`
	Color              string    `json:"color"`
	Weight             *float64  `json:"weight,omitempty"`
	ChipNumber         *string   `json:"chip_number,omitempty"`
	TattooNumber       *string   `json:"tattoo_number,omitempty"`
	PassportNumber     *string   `json:"passport_number,omitempty"`
	PhotoURL           *string   `json:"photo_url,omitempty"`
	Status             string    `json:"status"`
	VerificationStatus string    `json:"verification_status"`
	Sterilized         bool      `json:"sterilized"`
	SterilizationDate  *string   `json:"sterilization_date,omitempty"`
	SpecialMarks       *string   `json:"special_marks,omitempty"`
	Character          *string   `json:"character,omitempty"`
	HealthNotes        *string   `json:"health_notes,omitempty"`
	Allergies          *string   `json:"allergies,omitempty"`
	ChronicConditions  *string   `json:"chronic_conditions,omitempty"`
	EmergencyContact   *string   `json:"emergency_contact,omitempty"`
	EmergencyPhone     *string   `json:"emergency_phone,omitempty"`
	InsuranceCompany   *string   `json:"insurance_company,omitempty"`
	InsurancePolicy    *string   `json:"insurance_policy,omitempty"`
	City               *string   `json:"city,omitempty"`
	Region             *string   `json:"region,omitempty"`
	Urgent             *bool     `json:"urgent,omitempty"`
	ContactPhone       *string   `json:"contact_phone,omitempty"`
	ContactName        *string   `json:"contact_name,omitempty"`
	CreatedAt          time.Time `json:"created_at"`
	UpdatedAt          time.Time `json:"updated_at"`
}

// Treatment представляет обработку питомца (от блох, клещей, глистов)
type Treatment struct {
	ID         int        `json:"id"`
	PetID      int        `json:"pet_id"`
	UserID     int        `json:"user_id"`
	Date       time.Time  `json:"date"`
	Medication string     `json:"medication"`
	Dosage     string     `json:"dosage"`
	NextDate   *time.Time `json:"next_date,omitempty"`
	Notes      *string    `json:"notes,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
	UpdatedAt  time.Time  `json:"updated_at"`
}

// MedicalEvent представляет медицинское событие из PetID (от клиник)
type MedicalEvent struct {
	ID             int       `json:"id"`
	PetID          int       `json:"pet_id"`
	EventType      string    `json:"event_type"`
	UserID         int       `json:"user_id"`
	ClinicID       *int      `json:"clinic_id,omitempty"`
	OrganizationID *int      `json:"organization_id,omitempty"`
	EventDate      string    `json:"event_date"`
	Description    string    `json:"description"`
	Veterinarian   *string   `json:"veterinarian,omitempty"`
	NextVisitDate  *string   `json:"next_visit_date,omitempty"`
	CreatedAt      time.Time `json:"created_at"`
}

// StatusChangeRequest представляет запрос на изменение статуса питомца
type StatusChangeRequest struct {
	Status        string  `json:"status"`
	Date          string  `json:"date"`
	Location      *string `json:"location,omitempty"`
	Circumstances *string `json:"circumstances,omitempty"`
	ContactInfo   *string `json:"contact_info,omitempty"`
}

// TreatmentReminder представляет напоминание о предстоящей обработке
type TreatmentReminder struct {
	PetID       int    `json:"pet_id"`
	PetName     string `json:"pet_name"`
	TreatmentID int    `json:"treatment_id"`
	Medication  string `json:"medication"`
	NextDate    string `json:"next_date"`
	DaysUntil   int    `json:"days_until"`
	IsOverdue   bool   `json:"is_overdue"`
}

// User представляет пользователя
type User struct {
	ID        int       `json:"id"`
	Name      string    `json:"name"`
	LastName  string    `json:"last_name"`
	Email     string    `json:"email"`
	Avatar    string    `json:"avatar"`
	CreatedAt time.Time `json:"created_at"`
}

File: owner/frontend/types/owner.ts
// TypeScript types for Owner Cabinet

export interface Pet {
  id: number;
  owner_id: number;
  curator_id?: number;
  organization_id?: number;
  name: string;
  species: 'dog' | 'cat' | 'bird' | 'rodent' | 'reptile' | 'fish' | 'other';
  breed?: string;
  age: number;
  sex: 'male' | 'female';
  color: string;
  weight?: number;
  chip_number?: string;
  tattoo_number?: string;
  passport_number?: string;
  photo_url?: string;
  status: PetStatus;
  verification_status: 'pending_verification' | 'verified';
  sterilized: boolean;
  sterilization_date?: string;
  special_marks?: string;
  character?: string;
  health_notes?: string;
  allergies?: string;
  chronic_conditions?: string;
  emergency_contact?: string;
  emergency_phone?: string;
  insurance_company?: string;
  insurance_policy?: string;
  city?: string;
  region?: string;
  urgent?: boolean;
  contact_phone?: string;
  contact_name?: string;
  created_at: string;
  updated_at: string;
}

export type PetStatus = 'home' | 'lost' | 'found' | 'looking_for_home' | 'needs_help' | 'at_vet' | 'died';

export interface Treatment {
  id: number;
  pet_id: number;
  user_id: number;
  date: string;
  medication: string;
  dosage: string;
  next_date?: string;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface MedicalEvent {
  id: number;
  pet_id: number;
  event_type: 'registration' | 'vaccination' | 'sterilization' | 'treatment' | 'surgery' | 'examination';
  user_id: number;
  clinic_id?: number;
  organization_id?: number;
  event_date: string;
  description: string;
  veterinarian?: string;
  next_visit_date?: string;
  created_at: string;
}

export interface StatusChangeData {
  date: string;
  location?: string;
  circumstances?: string;
  contact_info?: string;
}

export interface TreatmentReminder {
  pet_id: number;
  pet_name: string;
  treatment_id: number;
  medication: string;
  next_date: string;
  days_until: number;
  is_overdue: boolean;
}

export interface User {
  id: number;
  email: string;
  name: string;
  avatar?: string;
}

// Form data types
export interface CreatePetData {
  name: string;
  species: Pet['species'];
  breed?: string;
  age: number;
  sex: Pet['sex'];
  color: string;
  weight?: number;
  chip_number?: string;
  tattoo_number?: string;
  passport_number?: string;
  sterilized?: boolean;
  sterilization_date?: string;
  special_marks?: string;
  character?: string;
  health_notes?: string;
  allergies?: string;
  chronic_conditions?: string;
  emergency_contact?: string;
  emergency_phone?: string;
  insurance_company?: string;
  insurance_policy?: string;
  city?: string;
  region?: string;
}

export interface CreateTreatmentData {
  pet_id: number;
  date: string;
  medication: string;
  dosage: string;
  next_date?: string;
  notes?: string;
}

export interface StatusChangeRequest {
  status: PetStatus;
  date: string;
  location?: string;
  circumstances?: string;
  contact_info?: string;
}

// API Response types
export interface ApiResponse<T> {
  data?: T;
  error?: string;
  message?: string;
}

export interface PetsResponse {
  pets: Pet[];
}

export interface TreatmentsResponse {
  treatments: Treatment[];
}

export interface MedicalHistoryResponse {
  events: MedicalEvent[];
}

export interface RemindersResponse {
  reminders: TreatmentReminder[];
}
